videostore/concat.go

```diff
@@ -112,6 +102,26 @@ func (c *concater) Concat(from, to time.Time, path string) error {
 	}
 }
 
+// writeConcatFileEntries writes the concat file entries to a file.
+func writeConcatFileEntries(entries []concatFileEntry, filePath string) error {
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	for _, entry := range entries {
+		lines := entry.string()
+		for _, line := range lines {
+			_, err := file.WriteString(line + "\n")
+			if err != nil {
+				return err
```

Comment:
lets delete the file if it fails.

use utils.NewGuard to

Reply:
Will look into `utils.NewGuard`. But couldn't I just move defer file deletion to above err check?